File: Mechanics_of_Materials-Computer-Problems/Code-Mechanics-of-Mat-Semester03/ali/chapter-2/p2_c1.py
import math

tel = int(input("Total Elements= "))
delta_point = 0

for c in range(1, tel + 1):
    length = float(input(f'length of element E:{c} (m)= '))
    diameter = float(input(f'diameter E:{c} (m)= '))
    elasticity = float(input(f'modulus of elasticity E:{c} (Pa)= '))
    load_applied = float(input(f'load-applied to its right/top E:{c} (N)= '))

    cs_area = (math.pi / 4) * diameter**2

    delta = (load_applied * length) / (cs_area * elasticity)
    delta_point += delta
    stress = load_applied / cs_area

    print(f'\n El {c} => Stress= {stress / 1e6} Pa, Delta= {delta * 1e3} mm, Delta Point= {delta_point * 1e3} mm\n')

File: Mechanics_of_Materials-Computer-Problems/Code-Mechanics-of-Mat-Semester03/kimya/chapter-2/p2_c1.m
clc; clear; close all;

tel = input("number of elements: ");
del_p = 0;

for i = 1:tel
    len = input(['length of element E:' num2str(i) ' (m): ']);
    dia = input(['diameter E:' num2str(i) ' (m): ']);
    m_el = input(['modulus of elasticity E:' num2str(i) ' (Pa): ']);
    l_a = input(['load-applied to its right/top E:' num2str(i) ' (N): ']);

    cs_area = (pi / 4) * (dia)^2;

    delta = (l_a * len) / (cs_area * m_el);
    del_p = del_p + delta;
    stress = l_a / cs_area;

    fprintf('\n     Element Number: %d ==> Stress: %f Pa, Delta: %f mm, Delta Point: %f mm \n', i, stress / 1e6, delta * 1e3, del_p * 1e3);
end